weblate/trans/util.py

```diff
@@ -35,6 +36,8 @@
     (140, gettext_lazy("Very low")),
 )
 
+CJK_PATTERN = re.compile(r"([ᄀ-ᇿ⺀-⿟⿰-鿿ꥠ-꥿가-퟿豈-﫿︰-﹏＀-￯𚿰-𛅯🈀-🋿\U00020000-\U0003FFFF]+)")
```

Comment:
@yheuhtozr Can you please verify wheter CodeQL is wrong here?

Reply:
Before judging whether it is wrong or not, I think the message is corrupted. `\ufffd` is REPLACEMENT CHARACTER (�) that suggests something went wrong during the emission of that error string. We might want to ask CodeQL to correctly print the code points they wanted to show, but my current hypotheses are:

- The range contains unassigned characters
  Obviously since that is what we did for performance to get consecutive ranges
- The range contains noncharacters
  Every last two code points in a Unicode plane are no-use characters. The pattern contains four in `\U00020000-\U0003FFFF` part. It'd be a tradeoff of potential security threats and performance drawback giving a more computationally complicated expression.

Reply:
Can we use `\uxxxx` instead of actual Unicode literals? I think that would make it easier to understand than this, as most people won't have fonts to display these characters.

Reply:
@nijel It is okay to do that, but does it solve the warning?

Reply:
Probably no, but it makes the code easier to read. We can ignore the warning as well, it's just a warning.